rfc6979/src/lib.rs

```diff
@@ -154,12 +158,21 @@ where
 
     /// Write the next `HMAC_DRBG` output to the given byte slice.
     pub fn fill_bytes(&mut self, out: &mut [u8]) {
-        for out_chunk in out.chunks_mut(self.v.len()) {
+        let mut out_chunks = out.chunks_exact_mut(self.v.len());
+
+        for out_chunk in &mut out_chunks {
             self.k.update(&self.v);
             self.v = self.k.finalize_reset().into_bytes();
             out_chunk.copy_from_slice(&self.v[..out_chunk.len()]);
         }
 
+        let out_remainder = out_chunks.into_remainder();
+        if !out_remainder.is_empty() {
+            self.k.update(&self.v);
+            self.v = self.k.finalize_reset().into_bytes();
+            out_remainder.copy_from_slice(&self.v[..out_remainder.len()]);
+        }
```

Comment:
Prior to #773 it was ensured that the output size would always be equal to the digest size, however now we need to handle the remainder.

https://datatracker.ietf.org/doc/html/rfc6979#appendix-A.1 contains a detailed example I need to step through and also add test cases for